pkg/util/create_or_update_test.go

```diff
@@ -156,6 +160,19 @@ var _ = Describe("CreateOrUpdate function", func() {
 			tests.EnsureNoActionsForResource(t.testingFake, "pods/status", "update")
 		})
 
+		Context("and GenerateName is set", func() {
+			BeforeEach(func() {
+				t.pod.Name = ""
+				t.pod.GenerateName = "name-prefix"
+				t.pod.Labels = map[string]string{"label1": "value1", "label2": "value2"}
+			})
+
+			It("should successfully create the resource", func() {
+				Expect(createOrUpdate(util.OperationResultCreated)).To(Succeed())
+				t.verifyPod()
+			})
+		})
```

Comment:
It would be worth checking that a name was generated and starts with the given prefix.

Reply:
Done